### PR TITLE
Extract config

### PR DESCRIPTION
**Pull request by alxwrd**
This extracts out the `Config` class into a seperate file to keep the `driver` file focused.

### CODE REVIEW

File: l293d/driver.py [hunk 1]
```diff
@@ -138,17 +39,12 @@ def get_pin_numbering(cls):
...
-    else:
-        print("pin_numbering must be either 'BOARD' or 'BCM'.")
-        raise ValueError("pin_numbering must be either 'BOARD' or 'BCM'.")
+    pin_num = Config.pin_numbering
+    v_print('Setting GPIO mode: {}'.format(pin_num))
+    GPIO.setmode(getattr(GPIO, pin_num))
+
 
-pins_in_use = []  # Lists pins in use (all motors)
+pins_in_use = Config.pins_in_use  # Lists pins in use (all motors)
```
Comment (alxwrd) [inline l293d/driver.py]: Creating a reference to the `list` in Config.

File: l293d/driver.py [hunk 2]
```diff
@@ -138,17 +39,12 @@ def get_pin_numbering(cls):
...
-    if Config.pin_numbering == 'BOARD':
-        v_print('Setting GPIO mode: BOARD')
-        GPIO.setmode(GPIO.BOARD)
-    elif Config.pin_numbering == 'BCM':
-        v_print('Setting GPIO mode: BCM')
-        GPIO.setmode(GPIO.BCM)
-    else:
-        print("pin_numbering must be either 'BOARD' or 'BCM'.")
-        raise ValueError("pin_numbering must be either 'BOARD' or 'BCM'.")
+    pin_num = Config.pin_numbering
```
Comment (alxwrd) [inline l293d/driver.py]: I've removed the checks here as the `Config` class won't allow `pin_numbering` to be anything other than "BOARD" or "BCM".

File: l293d/driver.py [hunk 3]
```diff
@@ -259,7 +155,6 @@ def remove(self):
         Remove motor
         """
         if self.exists:
-            global pins_in_use
```
Comment (alxwrd) [inline l293d/driver.py]: Dropping the global as it's not redefined.

File: tests/test_l293d.py [hunk 4]
```diff
@@ -84,6 +84,7 @@ def test_pin_numbering_lock(self):
                    'if motors already exist. Set this at '
                    'the start of your script.')
         m1.remove()
+        d.Config.pin_numbering = 'BOARD'
```
Comment (alxwrd) [inline tests/test_l293d.py]: I've had to reset this value in the test because I couldn't get `reload` to set the Config back to default.

**Comment by jamesevickery** [comment]
Good idea! Thanks :grin:

**Pull request merged**